job-archive: use statedir path if dbpath is not set

chu11 opened the pull request:
Per discussion in #4251.

- make `period` required in `job-archive` to make it do its thing
- if `dbpath` is not set, assume a path in `statedir` for job-archive, otherwise error
- in `rc1`, don't load `job-archive` module unless we're rank 0 & `archive.period` has been set

i have one commit below that is purely anal retentive, it can be dropped :-)

garlick commented on src/modules/content-files/content-files.c:
```diff
@@ -277,8 +277,10 @@ static struct content_files *content_files_create (flux_t *h)
     if (!(dbdir = flux_attr_get (h, "statedir")))
         dbdir = flux_attr_get (h, "rundir");
-    if (!dbdir)
+    if (!dbdir) {
         flux_log_error (h, "neither statedir nor rundir are set");
+        goto error;
+    }
```
Thanks for catching this!  Wow don't know how I managed to do that not once but twice.

garlick commented on etc/rc1:
```diff
@@ -30,7 +30,10 @@ modload 0 cron sync=heartbeat.pulse
-modload 0 job-archive
+period=`flux config get --default= archive.period`
+if test $RANK -eq 0 -a -n "${period}"; then
+    modload 0 job-archive
```
Since the rank has already been checked (and that's all `modload()` does), probably better to just call `flux module load job-archive` here.

garlick commented on src/modules/job-archive/job-archive.c:
```diff
@@ -543,10 +555,11 @@ int mod_main (flux_t *h, int ac, char **av)
 
-    /* we do nothing if no dbpath specified */
-    if (ctx->dbpath) {
+    /* we do nothing if period not set */
+    if (ctx->period > 0.0) {
         if (job_archive_init (ctx) < 0)
             goto done;
```
Shouldn't it be an error if the period is not set?  Is loading the module but having it do nothing a valid use case?

chu11 replied:
ahh yes, I was keeping the style that was previously there, but we should error out instead.

garlick reviewed (changes requested):
Thanks @chu11 - looking good.  Just a couple of things that I think need to be addressed and/or discussed.

chu11 commented:
re-pushed, addressing comments above.  the issues were pretty tiny so just squashed.  The check for a valid period was put into the `process_config()` function.

```
--- a/src/modules/job-archive/job-archive.c
+++ b/src/modules/job-archive/job-archive.c
@@ -544,6 +544,12 @@ static int process_config (struct job_archive_ctx *ctx)
         else
             ctx->busy_timeout = (int)(1000 * tmp);
     }
+
+    /* period is required to be set */
+    if (ctx->period == 0.0) {
+        flux_log_error (ctx->h, "period not set");
+        return -1;
+    }
     return 0;
 }
```

garlick reviewed (approved):
LGTM!  Thanks.

chu11 commented:
re-pushed, added 1 more test just trying to get more coverage.  If all is good I'll set MWP.

chu11 commented:
one more re-push with another test, we're getting closer to 80% :-) (but might not get there)

chu11 commented:
whew, hit 80% exactly
